src/main/java/org/jenkinsci/plugins/workflow/flow/BlockableResume.java

```diff
@@ -0,0 +1,15 @@
+package org.jenkinsci.plugins.workflow.flow;
+
+/**
+ * Can be added to advertise the ability to mark pipeline components which prevent pipelines from being able to resume
+ * after restart or after pause.
+ *
+ * Pipelines which cannot resume will simply fail.
+ */
+public interface BlockableResume {
+    /** Return true if we prevent the abiity to resume. */
+    public boolean isResumeBlocked();
+
+    /** Set resume on or off - may throw an {@link IllegalArgumentException} if trying to illegally toggle. */
```

Comment:
`IllegalStateException` is probably what you meant.

Under what conditions would it be “illegal” to toggle this? As discussed elsewhere, I suspect this could be deleted.